mmdet/apis/train.py

```diff
@@ -68,6 +68,51 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def scale_lr(cfg, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        logger (logging.Logger): logger.
+    """
+    enable_auto_scale_lr = cfg.get('enable_auto_scale_lr', False)
+    if enable_auto_scale_lr is False:
+        logger.info(f'You disabled automatically scaling LR.')
+        return
+
+    # default batch size of pre-train model
+    original_batch_size = cfg.get('default_batch_size', None)
+    assert original_batch_size is not None
+
+    # calculate the batch size
+    gpu_number = len(cfg.gpu_ids)
```

Comment:
We can use get_world_size() to get the full world size, which should be safe in multi-machine DDP.

Reply:
In file `tools\train.py`, when distributed enable, the `cfg.gpu_ids` will get info from `dist.get_world_size()`. And those steps process before function `scale_lr`. So it's safe to get the gpu number using `len(cfg.gpu_ids)` here.

![image](https://user-images.githubusercontent.com/25873202/158172899-34ae87cd-eb92-4953-842b-aebdb4410778.png)

![image](https://user-images.githubusercontent.com/25873202/158173097-2b44b71c-e23a-4313-90de-d02c8d0d3510.png)